daffodil-propgen/src/main/scala/org/apache/daffodil/propGen/PropertyGenerator.scala

```diff
@@ -642,12 +645,18 @@ object Currency {
    */
   def generateNonEnumStringPropInit(propName: String) = {
     val template =
-      """registerToStringFunction(()=>{getPropertyOption("currency") match {
+      """registerToStringFunction(()=>{getPropertyOption("currency", false) match {
         case None => ""
         case Some(value) => "currency='" + value.toString + "'"
       }
     })"""
-    val res = template.replaceAll("currency", propName)
+    val res = if (expressionAllowedProperties.contains(propName)) {
+      template
+        .replaceAll("currency", propName)
+        .replaceAll("false", "true") // add expression allowed for these
+    } else {
+      template.replaceAll("currency", propName)
+    }
```

Comment:
Thoughts on changing `false` to `expressionAllowed` and then just always do something like this?
```scala
val expressionAllowStr = expressionAllowedProperties.contains(propName).toString
template
  .replace("currency", propName)
  .replace("expressionAllowed", expressionAllowedStr)
```
Avoids the replace duplicate code and matches on something less common to avoid possible incorrect replacements if we ever tweak this template.